### PR TITLE
Re #2999: More strict SBUS frame check (only allow 25 byte frames, ch…

### PR DESCRIPTION
**Pull request by projectkk2glider**
…eck end byte)

This is what I had in mind when I made comment https://github.com/opentx/opentx/issues/2999#issuecomment-219238649

This code was not tested by me! I have no hardware for this at the moment.

### CODE REVIEW

**Comment by bsongis** [comment]
I will check it is OK after work

File: radio/src/sbus.cpp [hunk 1]
```diff
@@ -24,40 +24,31 @@
...
+  if (size != SBUS_FRAME_SIZE || sbus[0] != SBUS_START_BYTE || sbus[SBUS_FRAME_SIZE-1] != SBUS_END_BYTE) {
     return; // not a valid SBUS frame
   }
-
-  if (size < SBUS_MIN_FRAME_SIZE) {
-    return;
-  }
-
-  if (size > SBUS_FLAGS_IDX && ((sbus[SBUS_FLAGS_IDX] & (1<<SBUS_FAILSAFE_BIT)) || (sbus[SBUS_FLAGS_IDX] & (1<<SBUS_FRAMELOST_BIT)))) {
+  if (sbus[SBUS_FLAGS_IDX] & (1<<SBUS_FAILSAFE_BIT) || sbus[SBUS_FLAGS_IDX] & (1<<SBUS_FRAMELOST_BIT)) {
```
Comment (bsongis) [inline radio/src/sbus.cpp]: I would see parenthesis here. I am never sure which one gets the higher priority between & and ||

File: radio/src/sbus.cpp [hunk 2]
```diff
@@ -76,11 +67,15 @@ void processSbusInput()
...
+  static uint8_t SbusIndex = 0;
+  static uint16_t SbusTimer;
+  static uint8_t SbusFrame[SBUS_FRAME_SIZE];
+
   while (sbusGetByte(&rxchar)) {
     active = 1;
     SbusFrame[SbusIndex++] = rxchar;
-    if (SbusIndex > SBUS_MAX_FRAME_SIZE-1) {
-      SbusIndex = SBUS_MAX_FRAME_SIZE-1;
+    if (SbusIndex > SBUS_FRAME_SIZE-1) {
```
Comment (bsongis) [inline radio/src/sbus.cpp]: Then the size given to processSbusFrame will be 24?

**Comment by bsongis** [comment]
I just checked your changes with the fixes I did, and it seems OK

**Comment by projectkk2glider** [comment]
Cool, thanks!

**Pull request merged**